### ISSUE TITLE
Add various "esc" screen functionality and announcements

### ISSUE THREAD

**Issue by bhajneet** [ShabadOS/desktop#392]
Once there is a UI for #352 of long-pressing esc we can also add it as a right-click for esc button in the controller. 

This can give options to 'esc' the screen with something different like "Waheguru" slide, custom message slide (for moving illegally parked cars), custom image slide (for event posters), etc.

I'm guessing it's actually better to have a different method for custom messages as they may want to continue presenting the shabad alongside the message which is "stickied" in announcement colors/contrast at the top of the screen. However, I'm out of ideas on how to achieve this in our current UI. It could go into settings > display and not be a part of the "esc" UI. (Also stickied messages should not show up in livestreams, but non-image esc screens should probably show up). However I'm guessing they would want this "stickied" message to show up even on an escaped screen. So you'd have the announcement and a blank slide, waheguru slide, or poster image slide underneath / below the stickied message. So I don't think settings > display is the right location for this. Could be an exception to the rule. Could call the option "Urgent Announcement" with an "i" information which says "this field when toggled will not hide during esc". And a separate button for filling out a message for the announcement. This lets you turn on and off the same message if they're using it for something like "Sevadaars needed in the langar hall", the message can stay the same but easily toggled in settings.

@Harjot1Singh what are your thoughts? Should we have a completely different section in the UI so they can also have custom fields like "Dhan Guru Gobind Singh" which can be filled out by the user?

**Comment by bhajneet** [comment]
Will need to use a new icon which signifies "Slides" or something to do with custom projection/presentation

Current one looks like turning off projector

**Comment by Harjot1Singh** [comment]
I think anything that opens a new window/tab is too messy. Perhaps we can change the icon to be a new prsentation tab thing like you've said. If we want to add more details on how to use the remote controller without going to settings, do we want to also add it to the controller? Because then, maybe we should think about how to make this all fit, visually (short of moving to v3 UI)

**Comment by bhajneet** [comment]
The "parts" of this issue which I see:

1. An alternate to the blank "esc" screen which is also captured on live streams. E.g. waheguru slide or event poster. If it's just text, it can show up on live stream. If it's an image, I'm not so sure... We could perhaps set it so that the image takes up the whole screen, and in that case people may actually want it to show on live streams?
2. A method to set a "sticky" message above the presentation slide (whether bani or alt "esc" screen) which does not show up on the live stream. E.g. "Please move illegally parked car with license XYZ".

I envisioned setting the "esc" screen to be called a "break" screen. So it could be right clicked to "waheguru" which when the user hits esc will say waheguru. This setting persists across sessions and must be changed with a right click of the bottom right "esc/break" icon. One of these context menu items could be "custom image" and another could be "set new custom image" which opens a dialog for the user to pick an image. I think this is all doable with the current UI, though perhaps maybe a bit complicated and requiring dev time.

The second scenario should perhaps be broken into it's own new issue. It could be a separate icon (though our bottom bar is crowded) or it could be something that is toggled in settings. I prefer settings for the sticky message because it doesn't act like any of the other buttons we have in the controller. This button is more of a toggle. In settings it could say:

- Toggle sticky message (on/off)
- Sticky message text
- Priority red, yellow, green, blue (for some customization of the sticky message theme)
- Font size of sticky message
- Convert sticky message text to gurmukhi (if typing in ascii but wanting a punjabi message).

So for 2. I think a new area of settings would be best. If you have a sticky message you may want to turn off some display items (next line, punjabi translation) to save some space too. So to put it in settings, doesn't seem like a jarring step to me for sticky message purposes.

**Comment by bhajneet** [comment]
I forgot to mention for number 1. the esc screen, if people want to add their own variations like "Dhan Dhan Guru Granth Sahib ji" they should be able to with a alt-screen.json or such in user settings. Should be easily shareable. So a screen in settings for alt screen + sticky text might be nice with a button to open the alt screens folder for easy editing of the txt file.

**Comment by Harjot1Singh** [comment]
Add to right-click "Configure sticky message" etc

**Comment by Harjot1Singh** [comment]
Depends on #154
